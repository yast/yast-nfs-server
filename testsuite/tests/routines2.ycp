/**
 * Module:
 *   NFS server configuration
 *
 * Summary:
 *   Routines testsuite
 *
 * Authors:
 *   Martin Vidner <mvidner@suse.cz>
 *
 * $Id$
 */
{
    // testedfiles: routines.ycp Testsuite.ycp

    include "testsuite.ycp";
    include "nfs_server/routines.ycp";

    string test_options = "secure,insecure,rw,ro,sync,async,\
no_wdelay,wdelay,nohide,hide,no_subtree_check,subtree_check,\
insecure_locks,secure_locks,no_auth_nlm,auth_nlm,root_squash,\
no_root_squash,all_squash,no_all_squash";

    list<map<string,any> > exports = [$["allowed":["proj*.local.domain(rw)"], 
		      "mountpoint":"/projects"], 
		    $["allowed":["*.local.domain(ro)", "@trusted(rw)"], 
		      "mountpoint":"/usr"], 
		    $["allowed":["(ro,insecure,all_squash)"], 
		      "mountpoint":"/pub"]];

  TEST (``(CheckNoSpaces ("no_spaces")), [], nil);
  TEST (``(CheckNoSpaces (" space_before")), [], nil);
  TEST (``(CheckNoSpaces ("space inside")), [], nil);
  TEST (``(CheckNoSpaces (" space_before and inside")), [], nil);
  TEST (``(CheckNoSpaces ("space_after ")), [], nil);
  TEST (``(CheckNoSpaces (" before inside after ")), [], nil);
  TEST (``(CheckNoSpaces ("tab	ulator")), [], nil);

  TEST (``(CheckExportOptions ("rw,all_squash")), [], nil);
  TEST (``(CheckExportOptions ("invalid:\options <> ?")), [], nil);
  // colon is valid
  TEST (``(CheckExportOptions ("sec=none:kerb5")), [], nil);

  TEST (``(CheckExportOptions_strict ("ro,insecure,all_squash")), [], nil);
  TEST (``(CheckExportOptions_strict (test_options)), [], nil);
  TEST (``(CheckExportOptions_strict (test_options + ",some_invalid")), [], nil);
  TEST (``(CheckExportOptions_strict ("ro,invalid_option")), [], nil);
}
