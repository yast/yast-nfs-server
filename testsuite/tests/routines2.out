Return	true
Return	false
Return	false
Return	false
Return	false
Return	false
Return	false
Return	true
Return	false
Return	true
Return	true
Return	true
Return	false
Return	false
